Add search_guild_members

Kamisoi opened the pull request:
## What type of pull request is this?

<!-- Check whichever applies to your PR -->
- [x] Non-breaking code change
- [ ] Breaking code change
- [ ] Documentation change/addition 

## Description
<!-- Clearly and concisely describe what this PR is for, and why you feel it should be merged. -->
Added method for searching member of guild via string, also possible to get more than one by providing limit integer.

## Changes
<!-- - A bullet pointed list outlining the changes you have made -->
Add async method for [search_guild_members](https://discord.com/developers/docs/resources/guild#search-guild-members)

## Checklist

<!-- These are actions you **must** have taken, if you haven't, your PR will be rejected -->
- [x] I've formatted my code with [Black](https://black.readthedocs.io/en/stable/)
- [x] I've ensured my code works on `Python 3.10.x`
- [x] I've tested my code

Kamisoi committed:
Add search_guild_members

Kamisoi commented on dis_snek/models/discord_objects/guild.py:
```diff
@@ -1017,6 +1017,10 @@ async def get_guild_integrations(self) -> List["GuildIntegration"]:
         data = await self._client.http.get_guild_integrations(self.id)
         return [GuildIntegration.from_dict(d | {"guild_id": self.id}, self._client) for d in data]
 
+    async def search_guild_members(self, name: str, limit: int = 1) -> List["Member"]:
```
After some thought here it shouldn't be "name" but something else, since it can be any string that could be used to find user nickname or username.

LordOfPolls replied:
Query?

Kamisoi replied:
How about search_query, for even more clarification? (at least imo)

LordOfPolls replied:
Seems kinda redundant seeing as Query means to ask for information, and the method is literally called search

Kamisoi committed:
Change from `name` to `query`

Kamisoi committed:
Forgot to edit one `name` to `query`

My bad ._.

Kamisoi committed:
Change name from to `search_members`

From `search_guild_members` to `search_members`, since it's Guild method.
Suggested by https://github.com/artem30801
